src/common/transformations/src/transformations/common_optimizations/convert_pagedattn_inputs.cpp

```diff
@@ -0,0 +1,156 @@
+                                                          alibi_slopes,
+                                                          max_context_len});
+
+    auto pa_2 = makePattern<op::PagedAttentionExtension>({Q,
+                                                          K,
+                                                          V,
+                                                          key_cache_0,
+                                                          value_cache_0,
+                                                          past_lens,
+                                                          subsequence_begins,
+                                                          block_indices,
+                                                          block_indices_begins,
+                                                          scale,
+                                                          sliding_window,
+                                                          alibi_slopes,
+                                                          max_context_len,
+                                                          rotated_block_indices,
+                                                          rotation_deltas,
+                                                          rotation_trig_lut});
+    auto result = pa_1 | pa_2;
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto pa_op = m.get_match_root();
+        auto key_cache = ov::as_type_ptr<ov::op::v0::Parameter>(pa_op->get_input_node_shared_ptr(3));
+        auto value_cache = ov::as_type_ptr<ov::op::v0::Parameter>(pa_op->get_input_node_shared_ptr(4));
+        auto format_cache_precision = [](ov::element::Type cache_precision, ov::element::Type infer_precision) {
+            return cache_precision == ov::element::f16 && infer_precision == ov::element::bf16 ? infer_precision
+                                                                                               : cache_precision;
+        };
+        auto init_cache_shape = [&](const size_t head_nums,
+                                    const size_t head_size,
+                                    const size_t block_size,
+                                    const ov::element::Type precision,
+                                    const size_t group_size,
+                                    const bool bychannel,
+                                    const std::vector<size_t>& orders) {
+            size_t _block_size = block_size;
+            ov::Dimension::value_type _head_nums = head_nums;
+            ov::Dimension::value_type _head_size = head_size;
+            ov::Dimension::value_type _group_size = group_size;
+            _group_size = _group_size ? _group_size : _head_size;
+            if (!bychannel) {
+                if (_head_size % _group_size != 0) {
+                    OPENVINO_THROW("cache head_size ", head_size, "cannot be divided by group_size ", group_size);
```

Comment:
should we allow models execution without applying the transformation (`return false;`) or `_head_size % _group_size` definitely indicates that the model is incorrect?